### ISSUE TITLE
Doco on how to build a WasmEdge CLI 

### ISSUE THREAD

**Issue by sholtomaud** [WasmEdge/WasmEdge#1600]
## Motivation

I'm wanting to build a portable CLI with WasmEdge which I can run on my local machine, and possibly in a webcontainer in the browser. 

## Details

I was thinking that the WasmEdge doco on js would show me how to compile an CLI app as one option, a bit like how I can compile a nodejs app into a CLI (it is however really big).  But there doesn't seem to be any doco.

Can we have some doco on how to do this?

**Comment by juntao** [comment]
Do you mean to create a single binary app that contains WasmEdge and an Wasm app that runs inside it? You can use the host SDK to do that. Golang, Rust, and C are supported. 

https://wasmedge.org/book/en/embed.html

**Comment by sholtomaud** [comment]
That is exactly what I'm thinking about for javascript/nodejs yes. 

Or perhaps something like the ape format https://justine.lol/ape.html ... maybe

**Comment by juntao** [comment]
Perhaps you can create a demo on how to compile a C host + embedded JS app + embedded WasmEdge into a single ape binary. :)

**Comment by sholtomaud** [comment]
Yes please. If you could do that it would be awesome! And if you include some doco around it even better! 👍
